ledger/block/src/transaction/deployment/bytes.rs

```diff
@@ -70,6 +102,16 @@ impl<N: Network> ToBytes for Deployment<N> {
             // Write the certificate.
             certificate.write_le(&mut writer)?;
         }
+        // If the deployment version is 2, write the program checksum and program owner.
+        // Note: The unwraps are safe because `Deployment::version` only returns `V2` if both the checksum and owner is present.
+        if version == DeploymentVersion::V2 {
+            // Write the bytes of the checksum.
+            for byte in &self.program_checksum.unwrap() {
+                byte.write_le(&mut writer)?;
+            }
```

Comment:
wouldn't it be preferable to use `self.program_checksum.unwrap().write_le(&mut writer)?;` here instead? or do we deliberately want to omit the length of the checksum?